server/sublist.go

```diff
@@ -295,10 +319,18 @@ func (s *Sublist) Match(subject string) *SublistResult {
 
 	// Get result from the main structure and place into the shared cache.
 	// Hold the read lock to avoid race between match and store.
+	var n int32
+
 	s.RLock()
 	matchLevel(s.root, tokens, result)
-	s.cache.Store(subject, result)
-	n := atomic.AddInt32(&s.cacheNum, 1)
+	// Check for empty result.
+	if len(result.psubs) == 0 && len(result.qsubs) == 0 {
+		result = emptyResult
+	}
+	if ce != slNoCache {
```

Comment:
Consistent with check above (since we don't have the lock at beginning of function). But we could also simply check if `s.cache` is `nil` here.

Reply:
hmm good point, let me take a look.

Reply:
I like your idea, changed.

Reply:
Please take a quick look.